### PR TITLE
Track injection statistics for MCC

### PR DESCRIPTION
**Pull request by roelof-groenewald**
In this PR we add "flux" diagnostics for MCC to keep track of charge injected per step due to impact ionization.

I will make a couple upstream PRs to include a lot of these changes there but we can merge this whenever review is done if people want to start using this functionality (or test it to see if it is slow haha).

Example output from flux diag:
```
THIS DIAGNOSTIC PERIOD:
Cathode Electrons Current Emitted: -2.03891 A/cm^2
Cathode Electrons Current Collected: 0.000229378 A/cm^2
Cathode Electrons Current Net: -2.03869 A/cm^2
Cathode Ar_Ions Current Collected: 0 A/cm^2
Anode Electrons Current Collected: 2.05166 A/cm^2
Anode Ar_Ions Current Collected: -0.000127432 A/cm^2
mcc Electrons Current Emitted: -0.0119531 A/cm^2
mcc Ar_Ions Current Emitted: 0.0119531 A/cm^2
Total Current Emitted: -2.03891 A/cm^2
Total Current Collected: 2.05176 A/cm^2
Total Current Net: 0.0128452 A/cm^2
Total Power Net: -60.0893 W/cm^2
```

I still need to update the version number and changelog.

### CODE REVIEW

File: Source/Particles/WarpXParticleContainer.cpp [hunk 1]
```diff
@@ -798,7 +798,7 @@ Real WarpXParticleContainer::sumParticleCharge(bool local) {
     amrex::Real total_charge = 0.0;
 
     const int nLevels = finestLevel();
-    for (int lev = 0; lev < nLevels; ++lev)
+    for (int lev = 0; lev <= nLevels; ++lev)
```
Comment (KZhu-ME) [inline Source/Particles/WarpXParticleContainer.cpp]: Was the code always just skipping the finest level here?

**Review by KZhu-ME — APPROVED** [review]
Looks good, looking forward to using this later!

**Review by peterscherpelz — APPROVED** [review]
Looks good! You'll want to do an upstream PR of at least the MCCScattering.H comment and finestLevel() bugfix (assuming that's what it is per KZ's question). You could see about whether they would want the before/after collisions callbacks and/or python functions added upstream too.
Finally, if you can add MCC to the mwxp templates that would be good.
Thanks!

File: Source/Particles/WarpXParticleContainer.cpp (hunk 1 above)
Comment (roelof-groenewald) [reply Source/Particles/WarpXParticleContainer.cpp]: Yeap, that was a bug.

**Comment by roelof-groenewald** [comment]
@peterscherpelz Already ahead of you with upstream PRs 2790 - 2793 haha
I've also added MCC to the advanced cathode template already. I'll change the other templates in that same PR as well.

**Comment by roelof-groenewald** [comment]
Due to the above mentioned pending upstream PRs (which I believe should merge quickly), I will wait to merge this into `memaster` until I can also include those changes in an oveall `memaster` update to the latest development.

File: mewarpx/changelog.csv [hunk 2]
```diff
@@ -1,5 +1,10 @@
 Version, Physics version, Date,        List of changes
-6.0.4, 2, In progress, "
+6.1.0, 2, In progress, "
```
Comment (roelof-groenewald) [inline mewarpx/changelog.csv:2]:
```suggestion
6.1.0, 2, 1/26/2022, "
```

**Pull request merged**